rfcs/034-location_location_location/README.md

```diff
@@ -0,0 +1,403 @@
+
+    <table>
+      <tr><th>Sierra location</th><th># of items</th></tr>
+      <tr><td>bwith / bound in above</td><td>218</td></tr>
+      <tr><td>cwith / Contained in above</td><td>28</td></tr>
+      <tr><td>digit / Digitised Collections</td><td>10</td></tr>
+      <tr><td>wqrfe / Enquiry Line</td><td>2</td></tr>
+      <tr><td>temp1 / At Digitisation</td><td>1</td></tr>
+      <tr><td>wghig / Gallery</td><td>1</td></tr>
+      <tr><td>sgmip / sgmip</td><td>1</td></tr>
+      <tr><td>gblip / 215 Information Point</td><td>1</td></tr>
+    </table>
+
+    </details>
+
+    <details>
+    <summary>Proposed heuristic for matching Sierra holdings locations to controlled types</summary>
+
+    For Sierra holdings, the API only returns a location *code* (e.g. stax) via fixed field 40.
+
+    We will look up the location code to find the location name, then apply a similar rule to that which we apply for items.
+    If the location code or name are unrecognised, we will skip adding a location to the holding.
+
+    This heuristic allows us to create locations in our controlled type for **99.87% of all holdings**.
+    </details>
+
+*   We will add optional `"shelfLocation"` and `"shelfmark"` fields to the PhysicalLocation model.
+
+    These will only be populated if the locationType is `OpenShelves`.
+    The shelfLocation will be drawn from the location name in Sierra (e.g. "History of Medicine").
+    The shelfmark will be drawn from the callNumber field in Sierra (e.g. "/HIS").
+
+    For item records, these will be populated from the *location.name* and *callNumber* fields from the Sierra API response, respectively.
+    This gives us a good initial implementation, and we can refine it later if necessary.
+
+    For holdings records, only the `"shelfLocation"` will be populated.
+    The holdings records don't have enough information to populate a shelfMark.
+
+    In these cases, we will concatenate the shelfLocation and shelfmark to create the `"label"`.
+
+    <details>
+      <summary>Examples of the proposed output</summary>
+
+      An item in closed stores:
+
+      ```json
+      {
+        "id": "i1000008",
+        "label": "Closed stores",
+        "locationType": {
+          "code": "closed-stores",
+          "label": "Closed stores",
+          "type": "LocationType"
+        },
+        "type": "Location"
+      }
+      ```
+
+      An item on open shelves:
+
+      ```json
+      {
+        "id": "i1000011",
+        "label": "Medical Collection WP300 1902C96c",
+        "shelfLocation": "Medical Collection",
+        "shelfmark": "WP300 1902C96c",
+        "locationType": {
+          "code": "open-shelves",
+          "label": "Open shelves",
+          "type": "LocationType"
+        },
+        "type": "Location"
+      }
+      ```
+
+      An online journal resource:
+
+      ```json
+      {
+        "id": "h1234567",
+        "url": "https://journals.example.org/login",
+        "link_text": "View this journal",
```

Comment:
Arguably a `label` property could be reused here, but I'm happy to add another more specific one.

Reply:
Yeah, I went back and forth on this. In the end, I made it a separate property because the label feels like something that should stand alone (e.g. for an aggregation), whereas this text doesn't make sense without the accompanying URL.

Reply:
Makes sense